pkg/diagnose/connectivity/hooks.go

```diff
@@ -0,0 +1,118 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package connectivity contains logic for connectivity troubleshooting between the Agent
+// and Datadog endpoints. It uses HTTP request to contact different endpoints and displays
+// some results depending on endpoints responses, if any.
+package connectivity
+
+// This file contains all the functions used by httptrace.ClientTrace.
+// Each function is called at a specific moment during the communication
+// Their prototypes are defined by htpp.Client so variables might be unused
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http/httptrace"
+
+	"github.com/fatih/color"
+)
+
+// During a request, the http.Client will call the functions of the ClientTrace at specific moments
+// This is useful to get extra information about what is happening and if there are errors during
+// connection establishment, DNS resolution or TLS handshake for instance
+var DiagnoseTrace = &httptrace.ClientTrace{
```

Comment:
`DiagnoseTrace` don't need to be exported as it is only used in the same package (`connectivity`).

Reply:
Done